car_showroom/tasks.py

```diff
@@ -0,0 +1,77 @@
+from internship_crm.celery import app
+
+from .models import CarShowroom, CarShowroomCar
+from provider.models import Car, ProviderCar, ProviderSale
+
+
+@app.task
+def car_showroom_buy_provider():
+    for car_showroom in CarShowroom.objects.all():
+        print(car_showroom)
+        specs = car_showroom.specs
+        try:
+            specs = specs['where_to_buy']
+            for car_id, provider_id in specs.items():
+                provider_car = ProviderCar.objects.get(car_id=int(car_id),
+                                                       provider_id=provider_id)
+                if car_showroom.balance >= provider_car.price:
+                    try:
+                        car_showroom_car = CarShowroomCar.objects.get(car_showroom=car_showroom,
+                                                                      car=provider_car.car)
+                        car_showroom_car.value += 1
+                        car_showroom_car.save()
+                    except CarShowroomCar.DoesNotExist:
+                        CarShowroomCar.objects.create(car_showroom=car_showroom,
+                                                      car=provider_car.car,
+                                                      price=provider_car.price * 1.1,
+                                                      value=1)
+                    car_price = provider_car.price
+                    try:
+                        provider_sale = ProviderSale.objects.get(cars_list=int(car_id),
+                                                                 provider=provider_id,
+                                                                 is_active=True)
+                        car_price = car_price * (1 - (provider_sale.discount / 100))
+                    except ProviderSale.DoesNotExist:
+                        print('sale does not exists(')
+                    car_showroom.balance -= car_price
+                    car_showroom.save()
+                else:
+                    print('not enough money, bitch!')
+        except KeyError:
```

Comment:
Remove KeyError handling using dict.get() method to get values safely. nested try .. except blocks are not good